### PR TITLE
Improve Fetch incidents structure and add append notes funcitonality

### PR DESCRIPTION
**Pull request by DNRRomero**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by DNRRomero** [comment]
Hi @merit-maita  we've had some issues with setting the release notes, as the pipeline shows internal errors
could you give us a hand?

File: Packs/ZeroFox/Integrations/ZeroFox/ZeroFox.py [hunk 1]
```diff
@@ -1739,34 +1707,6 @@ def send_alert_attachment_command(
     )
 
 
-def get_alert_attachments_command(
```
Comment (merit-maita) [inline Packs/ZeroFox/Integrations/ZeroFox/ZeroFox.py:1742]: no need to remove it's code, you can just add the deprecated key in the yml file

File: Packs/ZeroFox/Integrations/ZeroFox/ZeroFox.py [hunk 2]
```diff
@@ -1940,7 +1880,6 @@ def main():
         "zerofox-modify-alert-notes": modify_alert_notes_command,
         "zerofox-submit-threat": submit_threat_command,
         "zerofox-send-alert-attachment": send_alert_attachment_command,
-        "zerofox-get-alert-attachments": get_alert_attachments_command,
```
Comment (merit-maita) [inline Packs/ZeroFox/Integrations/ZeroFox/ZeroFox.py:1943]: same as previous comment

File: Packs/ZeroFox/Integrations/ZeroFox/ZeroFox.yml [hunk 3]
```diff
@@ -1350,17 +1358,6 @@ script:
     - name: entry_id
       description: The entry ID of the attachment.
       required: true
-  - name: zerofox-get-alert-attachments
```
Comment (merit-maita) [inline Packs/ZeroFox/Integrations/ZeroFox/ZeroFox.yml:1353]: please add deprecated key without removing:
deprecated: true

File: Packs/ZeroFox/ReleaseNotes/1_3_1.json [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
+    "breakingChanges": true,
+    "breakingChangesNotes": "The zerofox-get-alert-attachments command was deleted to avoid confusion from customers on its usage."
```
Comment (merit-maita) [inline Packs/ZeroFox/ReleaseNotes/1_3_1.json:3]: please use the same comment in the rn, it's more clear
```suggestion
    "breakingChangesNotes": "Deprecated the ***zerofox-get-alert-attachments*** command due to unintended behavior."
```

File: Packs/ZeroFox/ReleaseNotes/1_3_1.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
+
+#### Integrations
+
+##### ZeroFox
+
+- Deprecated the `zerofox-get-alert-attachments` command due to unintended behavior.
+- Improved implementation of `fetch-incidents` for reducing api calls.
+- Added the `action` parameter to `zerofox-modify-alert-notes` command to include append functionality.
```
Comment (merit-maita) [inline Packs/ZeroFox/ReleaseNotes/1_3_1.md:8]:
```suggestion
- Deprecated the ***zerofox-get-alert-attachments*** command due to unintended behavior.
- Improved implementation of ***fetch-incidents*** for reducing api calls.
- Added the **action** parameter to ***zerofox-modify-alert-notes*** command to include append functionality.
```

**Review by merit-maita — COMMENTED** [review]
Thank you for your contribution!
please see my comments, and let's set up a short demo

**Comment by merit-maita** [comment]
> Hi @merit-maita we've had some issues with setting the release notes, as the pipeline shows internal errors could you give us a hand?

it's fine, the rn validator is expected to fail.

**Comment by DNRRomero** [comment]
Hi @merit-maita  we can give you a demo of the changes whenever you prefer between 8am and 4pm ET

**Comment by merit-maita** [comment]
@DNRRomero  can we set the demo for Monday 8:00 am ET?

**Comment by DNRRomero** [comment]
@merit-maita I did not see this message on time, as we start at 8 am ET, can we do it tomorrow (tuesday) at 10am ET?

**Comment by merit-maita** [comment]
> @merit-maita I did not see this message on time, as we start at 8 am ET, can we do it tomorrow (tuesday) at 10am ET?

i have meeting at 10:00 ET, maybe 9:30 ET?

**Comment by dramirez-SR** [comment]
@merit-maita 9:30 works for me!

**Comment by merit-maita** [comment]
> @merit-maita 9:30 works for me!

because of the time difference, we always miss setting up the meeting, so please once you see this comment, set up a meeting at 9:30, today or tomorrow. my email is mmaayta@paloaltonetworks.com

**Comment by merit-maita** [comment]
@dramirez-SR the meeting has no link to a zoom or teams meeting.

**Review by merit-maita — APPROVED** [review]

**Comment by merit-maita** [comment]
@DNRRomero thank you for the demo!
your pr has a merge conflict, please resolve it so i can go ahead and merge it.

**Comment by DNRRomero** [comment]
@merit-maita  Hi Merit! merge conflict has been resolved

**Pull request merged**